File: archive/ruby/muse/algorithms/sorting/SelectionSort.rb
module SelectionSort
  def self.sort(arr)
    return if arr.empty?

    iMin = 0

    bound = arr.length - 1
    0.upto(bound - 1) do |i|
      iMin = i
      (i + 1).upto(bound) do |j|
        iMin = j if arr[j] < arr[iMin]
      end
      arr[i], arr[iMin] = arr[iMin], arr[i] if iMin != i
    end
  end
end

File: archive/nodejs/muse/algorithms/sorting/SelectionSort.js
'use strict'

exports.sort = (arr) => {
  if (arr.length === 0) {
    return
  }

  let iMin = 0

  for (let i = 0, bound = arr.length - 1; i < bound; i++) {
    iMin = i
    for (let j = i + 1; j <= bound; j++) {
      if (arr[j] < arr[iMin]) {
        iMin = j
      }
    }
    if (iMin !== i) {
      [arr[i], arr[iMin]] = [arr[iMin], arr[i]]
    }
  }
}
